ddtrace/bootstrap/preload.py

```diff
@@ -66,11 +66,17 @@ def register_post_preload(func: t.Callable) -> None:
 
     symbol_db.bootstrap()
 
-if di_config.enabled or ed_config.enabled:
+# TODO: Do not enable RC features if not required
+if di_config.enabled or ed_config.enabled or config._trace_span_origin_enabled:
     from ddtrace.debugging import DynamicInstrumentation
 
     DynamicInstrumentation.enable()
```

Comment:
enable DynamicInstrumentation.enable() without di_config.enabled still get probes?
I wonder if we need to split product features (get RC probes) from the library capabilities features (instrument methods/lines)